### ISSUE TITLE
Allow for scenario sequencing in YAML

### ISSUE THREAD

**Issue by jshook** [nosqlbench/nosqlbench#67]
We need a way to allow scenarios to run with few details from a user. A single command should be enough to kick off a workload that has multiple phases. A proposed yaml format is below:

A new root-level property `scenario` is suggested for the YAML format. This property is a way to specify a complete testing workflow that can take the place of the user having to provide the commands on the command line.The contents of the `scenario` property is a list of scenario commands.

When a scenario property is provided in the YAML, it is *ignored* if a user uses the long-form syntax. Specifically, the only way to take advantage of scenario commands is to use one of two command line forms:

```
# fully controlled by a default or the only workflow
nb myfile.yaml
# this can also be abbrevated to
nb myfile

# a specified named workflow is used
nb myfile.yaml workflow1
# this can also be abbreviated to
nb myfile workflow1
```

In either case, a user can provide additional parameters after the scenario and workflow.

Scenarios are only passed as a map of lists. This is a way to have multiple workflows to pick from, and each name represents its own workflow.

- Workflows which have commands in the form of `[name]` automatically include the named workflow in place. Specifically, the YAML parser is responsible for duplicating the named workflow in place. This is a simple way to keep workflows DRY.

Since the scenario is contained within a YAML, it is not necessary for the yaml parameter to be provided in the scenario commands. It is valid, however, to provide one, as scenarios may often depend on multiple YAML files. However, scenario command interpretation *only* occurs at the highest level. 

All standard commands are supported. However, scripting commands are evaluated in the self-same scenario scripting environment as the top level. There are no nesting scopes for scripting.

Examples:
```
# single scenario command, fully automatic with
# nb myyaml
scenarios:
 default:
   - run type=cql ...
```

``` 
# named workflows, no default provided.
# user is prompted for one of the named workflows
scenarios:
 schema:
  - run type=TEMPLATE(type,cql) yaml=THIS host=TEMPLATE(host,dsehost) tags=phase:schema
 rampup:
  - run type=TEMPLATE(type,cql) .... tags=phase:rampup
 main:
  - run type=TEMPLATE(type,cql) ... tags=phase:main
```

```
# a default scenario is provided. The user will not be prompted, but they may override the
# workflow selection
scenarios:
 schema:
  - run type=TEMPLATE(type,cql) yaml=THIS host=TEMPLATE(host,dsehost) tags=phase:schema
 default:
  - [schema]
  - run type=TEMPLATE(type,cql) host=TEMPLATE(host,dsehost) tags=phase:rampup 
   cycles=TEMPLATE(cycles,1M)
```

**Comment by phact** [comment]
Can we use `scenario` instead of `workflow` for consistency?

**Comment by phact** [comment]
This has been addressed in #86 and #75

**Comment by phact** [comment]
Some unit tests related to this: 

https://github.com/nosqlbench/nosqlbench/blob/issue-84/engine-cli/src/test/java/io/nosqlbench/engine/cli/TestNBCLIOptions.java#L196-L222

TL;DR - yamls that have a scenario section as outlined above will be accessible via:

./nb <workload_name> [<scenario_name_filter>] [<param_key>=<param_value>]
